code/modules/reagents/chemistry/reagents/pyrotechnic.dm

```diff
@@ -203,8 +203,6 @@
 			S.create_reagents(volume)
 		S.reagents.add_reagent("thermite", volume)
 		S.thermite = TRUE
-		S.overlays.Cut()
-		S.overlays = image('icons/effects/effects.dmi', icon_state = "thermite")
```

Comment:
The first line here is the troublemaker. All it really does currently is just remove wall smoothing if you pour thermite on a wall, and does nothing on a floor.

The second line here doesn't even work, and even if it did, it wouldn't matter because thermiting a wall/floor burns and deletes it completely.